public/app/features/dashboard/components/SubMenu/SubMenu.tsx

```diff
@@ -51,10 +51,6 @@ class SubMenuUnConnected extends PureComponent<Props> {
 
     const styles = getStyles(theme);
 
-    if (!dashboard.isSubMenuVisible()) {
-      return null;
-    }
-
```

Comment:
Since it is only used in DashboardPage, I'd figured moving the logic there instead made more sense.